src/app/modules/attachment/attachment.service.ts

```diff
@@ -0,0 +1,51 @@
+import crypto from 'crypto'
+import { ResultAsync } from 'neverthrow'
+
+import { aws as AwsConfig } from '../../../config/config'
+import { createLoggerWithLabel } from '../../../config/logger'
+import { AttachmentUploadError } from '../core/core.errors'
+
+const logger = createLoggerWithLabel(module)
+
+export const uploadAttachments = (
+  formId: string,
+  attachmentData: Record<string, unknown>,
+): ResultAsync<Map<string, string>, AttachmentUploadError> => {
+  const attachmentMetadata = new Map()
+  const attachmentUploadPromises = []
+
+  for (const fieldId in attachmentData) {
```

Comment:
this is a matter of personal preference, but you could consider using `Object.keys(attachmentData).reduce(...)` rather than a `for` loop, to make this code more maintainable. the accumulated data might look something like this:
```
import { ManagedUpload } from 'aws-sdk/clients/s3'

type AttachmentReducerData = {
  attachmentMetadata: Map<string, string>
  attachmentUploadPromises: Promise<ManagedUpload.SendData>[]
}
```

but again this is subjective, since arguably a `reduce` might be less readable.

Reply:
Ah my preference is usually towards imperative style programming, but sure that works. Changed.